README.rst

```diff
@@ -0,0 +1,281 @@
+
+Internals
+---------
+
+There are three main parts of C++ connector: IO-zero-copy buffer,
+msgpack encoder/decoder and client handling requests itself.
+
+Buffer
+~~~~~~
+
+Buffer is parametrized by allocator, which means that users are able to
+choose which allocator will be used to provide memory for buffer's
+blocks. Data is originized into linked list of blocks of fixed size which
+is specified as template parameter of buffer.
+
+Client API
+~~~~~~~~~~
+
+**TODO: see src/Client/Connection.hpp and src/Client/Connector.hpp**
+
+Build
+-----
+
+To build C++ connector from sources one should have pre-installed
+compiler complied with C++17 standard, cmake/make utilities. Currently
+supported OS is Linux.
+
+Steps to build:
+
+.. code-block:: bash
+
+   cmake .
+   make
+
+Usage
+-----
+
+Embedding
+~~~~~~~~~
+
+Connector can be embedded in any C++ application with including main
+header: ``#include "<path-to-cloned-repo>/src/Client/Connector.hpp"``
+
+Objects instantiation
+~~~~~~~~~~~~~~~~~~~~~
+
+To create client one should specify buffer's and network provider's
+implementations as template parameters. Connector's main class has the
+following signature:
+
+.. code-block:: c
+
+   template<class BUFFER, class NetProvider = DefaultNetProvider<BUFFER>>
+   class Connector;
+
+If one don't want to bother with implementing its own buffer or network
+provider, one can use default one: ``tnt::Buffer<16 * 1024>`` and
+``DefaultNetProvider<tnt::Buffer<16 * 1024>>``. So the default
+instantiation would look like:
+
+.. code-block:: c
+
+   using Buf_t = tnt::Buffer<16 * 1024>;
+   using Net_t = DefaultNetProvider<Buf_t >;
+   Connector<Buf_t, Net_t> client;
+
+Client itself is not enough to work with Tarantool instances, so let's
+also create connection objects. Connection takes buffer and network
+provider as template parameters as well (note that they must be the same
+as ones of client):
+
+.. code-block:: c
+
+   Connection<Buf_t, Net_t> conn(client);
+
+Connecting
+~~~~~~~~~~
+
+Now assume Tarantool instance is listening ``3301`` port on localhost.
+To connect to the server we should invoke ``Connector::connect()``
+method of client object and pass three arguments: connection instance,
+address and port.
+``int rc = client.connect(conn, address, port)``.
+
+Error handling
+~~~~~~~~~~~~~~
+
+Implementation of connector is exception free, so we rely on return
+codes: in case of fail, ``connect()`` will return ``rc < 0``. To get
+error message corresponding to the last error happened during
+communication with server, we can invoke ``Connection::getError()``
+method:
+
+.. code-block:: c
+
+   if (rc != 0) {
+       assert(conn.status.is_failed);
+       std::cerr << conn.getError() << std::endl;
+   }
+
+To reset connection after errors (clean up error message and connection
+status), one can use ``Connection::reset()``.
+
+Preparing requests
+~~~~~~~~~~~~~~~~~~
+
+To execute simplest request (i.e. ping), one can invoke corresponding
```

Comment:
Does "i.e." mean ["that is"](https://www.merriam-webster.com/dictionary/i.e.) or "for example"?

```suggestion
To execute a simplest request (for example, ping), one can invoke corresponding
```